hoomd/hpmc/compute.py

```diff
@@ -7,123 +7,85 @@
+
+
+    Attributes:
+        test_particle_type (str): Type of particle to use when
+            computing free volume
+
+        num_samples (int): Number of samples to use in MC
+            integration
 
     """
-    def __init__(self, mc, seed, suffix='', test_type=None, nsample=None):
-
-        # initialize base class
-        _compute.__init__(self);
-
-        # create the c++ mirror class
-        cl = _hoomd.CellList(hoomd.context.current.system_definition);
-        hoomd.context.current.system.addCompute(cl, "auto_cl3")
-
-        cls = None;
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            if isinstance(mc, integrate.sphere):
-                cls = _hpmc.ComputeFreeVolumeSphere;
-            elif isinstance(mc, integrate.convex_polygon):
-                cls = _hpmc.ComputeFreeVolumeConvexPolygon;
-            elif isinstance(mc, integrate.simple_polygon):
-                cls = _hpmc.ComputeFreeVolumeSimplePolygon;
-            elif isinstance(mc, integrate.convex_polyhedron):
-                cls = _hpmc.ComputeFreeVolumeConvexPolyhedron;
-            elif isinstance(mc, integrate.convex_spheropolyhedron):
-                cls = _hpmc.ComputeFreeVolumeSpheropolyhedron;
-            elif isinstance(mc, integrate.ellipsoid):
-                cls = _hpmc.ComputeFreeVolumeEllipsoid;
-            elif isinstance(mc, integrate.convex_spheropolygon):
-                cls =_hpmc.ComputeFreeVolumeSpheropolygon;
-            elif isinstance(mc, integrate.faceted_ellipsoid):
-                cls =_hpmc.ComputeFreeVolumeFacetedEllipsoid;
-            elif isinstance(mc, integrate.polyhedron):
-                cls =_hpmc.ComputeFreeVolumePolyhedron;
-            elif isinstance(mc, integrate.sphinx):
-                cls =_hpmc.ComputeFreeVolumeSphinx;
-            elif isinstance(mc, integrate.convex_spheropolyhedron_union):
-                cls = _hpmc.ComputeFreeVolumeConvexPolyhedronUnion
-            elif isinstance(mc, integrate.faceted_ellipsoid_union):
-                cls = _hpmc.ComputeFreeVolumeFacetedEllipsoidUnion
-            elif isinstance(mc, integrate.sphere_union):
-                cls = _hpmc.ComputeFreeVolumeSphereUnion;
+    def __init__(self, test_particle_type, num_samples):
+        # store metadata
+        param_dict = ParameterDict(
+            test_particle_type=str,
+            num_samples=int
+        )
+        param_dict.update(
+            dict(test_particle_type=test_particle_type,
+                 num_samples=num_samples))
+        # set defaults
+        self._param_dict.update(param_dict)
+
+    def _attach(self):
+        integrator = self._simulation.operations.integrator
+        if not isinstance(integrator, integrate.HPMCIntegrator):
+            raise RuntimeError("The integrator must be an HPMC integrator.")
+
+        # Extract 'Shape' from '<hoomd.hpmc.integrate.Shape object>'
+        integrator_name = str(integrator).split()[0].split('.')[-1]
```

Comment:
You could just use this for the same effect.
```suggestion
        integrator_name = integrator.__class__.__name__
```